node/src/command.rs

```diff
@@ -118,18 +141,34 @@ fn extract_genesis_wasm(chain_spec: &Box<dyn sc_service::ChainSpec>) -> Result<V
 macro_rules! construct_async_run {
 	(|$components:ident, $cli:ident, $cmd:ident, $config:ident| $( $code:tt )* ) => {{
 		let runner = $cli.create_runner($cmd)?;
-		runner.async_run(|$config| {
-			let $components = new_partial::<
-				RuntimeApi,
-				TemplateRuntimeExecutor,
-				_
-			>(
-				&$config,
-				crate::service::parachain_build_import_queue,
-			)?;
-			let task_manager = $components.task_manager;
-			{ $( $code )* }.map(|v| (v, task_manager))
-		})
+		let chain_spec = &runner.config().chain_spec;
+		if chain_spec.is_turing() {
+			runner.async_run(|$config| {
```

Comment:
can we put some of these functions in a method under runner that's like async_turing_run and async_neumann_run instead? would clean up the code quite a bit